File: TaskManager/Lab_final_210042166/Task_Management_System/Issue_Repository/IssueRepository.java
package Lab_final_210042166.Task_Management_System.Issue_Repository;
import Lab_final_210042166.Task_Management_System.Issue.Issue;

import java.util.HashMap;
import java.util.Map;


public class IssueRepository {
    private static IssueRepository instance;
    private Map<String, Issue> issues;

    private IssueRepository() {
        issues = new HashMap<>();
    }

    public static IssueRepository getInstance() {
        if (instance == null) {
            instance = new IssueRepository();
        }
        return instance;
    }

    public void addIssue(Issue issue) {
        issues.put(issue.getId(), issue);
    }

    public Issue getIssue(String id) {
        return issues.get(id);
    }

    public void updateIssue(Issue issue) {
        issues.put(issue.getId(), issue);
    }

    public void removeIssue(String id) {
        issues.remove(id);
    }
}

File: TaskManager/LabFinal/LabFinal/IssueRepository.cs
namespace LabFinal
{
    public class IssueRepository
    {
        private readonly Dictionary<int, Issue> _issues = new Dictionary<int, Issue>();

        public void AddIssue(Issue issue)
        {
            _issues[issue.Id] = issue;
        }

        public void RemoveIssue(int issueId)
        {
            _issues.Remove(issueId);
        }

        public Issue GetIssue(int issueId)
        {
            return _issues[issueId];
        }
    }
}